### PR TITLE
Update csharp plugin to default to --v3

### LINKED ISSUE #3849: Make autorest --csharp default to track2
Currently `autorest --chsharp` default to [V2 version of C# generator](https://github.com/Azure/autorest/blob/master/packages/extensions/core/resources/plugin-csharp.md) and uses "@microsoft.azure/autorest.csharp" package. Now that we added command line support for track-2 C# generator, we should update the default to V3 C# generator and use "@autorest/csharp" package instead.

We can have --legacy flag as [Java](https://github.com/Azure/autorest/blob/master/packages/extensions/core/resources/plugin-java.md) to enable the use of the V2 C# generator and for this we can use `2.3.91` preview version of "@microsoft.azure/autorest.csharp" package.

### PR DESCRIPTION
**Pull request by timotheeguerin**
fix #3849

### CODE REVIEW

File: packages/extensions/core/resources/plugin-csharp.md [hunk 1]
```diff
@@ -2,17 +2,25 @@
 
 The V2 version of the C# Generator.
 
-``` yaml $(csharp) && $(preview) && !isRequested('@autorest/csharp')
-# default the v2 generator to using the last stable @microsoft.azure/autorest-core 
+``` yaml $(java) && !$(legacy) && !$(v2) && !isRequested('@microsoft.azure/autorest.csharp')
```
Comment (ShivangiReja) [inline packages/extensions/core/resources/plugin-csharp.md:5]:
```suggestion
``` yaml $(csharp) && !$(legacy) && !$(v2) && !isRequested('@microsoft.azure/autorest.csharp')
```

File: packages/extensions/core/resources/plugin-csharp.md [hunk 2]
```diff
@@ -2,17 +2,25 @@
 
 The V2 version of the C# Generator.
 
-``` yaml $(csharp) && $(preview) && !isRequested('@autorest/csharp')
-# default the v2 generator to using the last stable @microsoft.azure/autorest-core 
+``` yaml $(csharp) && !$(legacy) && !$(v2) && !isRequested('@microsoft.azure/autorest.csharp')
+version: ~3.0.6298
```
Comment (ShivangiReja) [inline packages/extensions/core/resources/plugin-csharp.md:6]: Where did you get `3.0.6298` version? We should use [@autorest/csharp@3.0.0-beta.20210205.1](https://www.npmjs.com/package/@autorest/csharp/v/3.0.0-beta.20210205.1) version or above.

File: packages/extensions/core/resources/plugin-csharp.md [hunk 3]
```diff
@@ -2,17 +2,25 @@
...
+
+use-extension:
+  "@autorest/csharp": "^3.0.0"
```
Comment (ShivangiReja) [inline packages/extensions/core/resources/plugin-csharp.md:9]: We should "latest" version of [@autorest/csharp](https://www.npmjs.com/package/@autorest/csharp) like [python](https://github.com/Azure/autorest/blob/master/packages/extensions/core/resources/plugin-python.md). Right @pakrym ?

File: packages/extensions/core/resources/plugin-csharp.md (hunk 2 above)
Comment (ShivangiReja) [inline packages/extensions/core/resources/plugin-csharp.md:6]: What is `~3.0.6298` version used for?

Comment (timotheeguerin) [reply packages/extensions/core/resources/plugin-csharp.md:6]: That's autorest core version

File: packages/extensions/core/resources/plugin-csharp.md (hunk 3 above)
Comment (pakrym) [reply packages/extensions/core/resources/plugin-csharp.md:9]: Yeah, I think we should be consistent.

**Review by daviwil — APPROVED** [review]

**Pull request merged**